ProcessList.c

```diff
@@ -592,7 +606,7 @@ Process* ProcessList_getProcess(ProcessList* this, pid_t pid, bool* preExisting,
    Process* proc = (Process*) Hashtable_get(this->processTable, pid);
    *preExisting = proc != NULL;
    if (proc) {
-      assert(Vector_indexOf(this->processes, proc, Process_pidCompare) != -1);
+      assert(ProcessList_indexOf(this, proc) != -1);
```

Comment:
The `indexOf` function is intentionally left with the `Vector` class.

A good optimizer will inline `Process_pidCompare` and optiize the resulting function to effectively what you wrote by hand. No need to maintain separate copies of this code for each scenario.

Reply:
Thanks for the quick review!

This commit removes `Vector_indexOf` because it was only used in `ProcessList.c` and only for this one exact task. 

Happy to add it back, but without Link Time Optimization GCC/Clang cannot optimize across translations units AFAIK and it was not inlined here (checked on Linux with gcc 10.3.0). Moving `Vector_indexOf` to `Vector.h` would allow the compiler to potentially inline the comparison function (but no guarantees).

Reply:
There was a subtle issue with your moved implementation: It lacked the consistency check that the original version had. But apart from this: I'm aware that LTO is required for a good optimizer to do its job™, but as things are now there'S quite a few other things that will tackle more runtime then trying to gain some linear speedup from having the optimizer work harder on a bad algorithm. ;-)

Reply:
Sounds good and I added back `Vector_indexOf` and removed `ProcessList_indexOf`. 

Also for future reference, clang will inline both `Vector_indexOf` and the comparison function if both are defined `static inline ...` in their respective header files. I left `Vector_indexOf` in `Vector.c` as it is used only once and putting it in the header would require exposing `Vector_isConsistent`, which seems unnecessary.